hoot-core/src/main/cpp/hoot/core/visitors/geometrymodifiers/PolyToWayGeoModifierAction.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. Maxar
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2019, 2020, 2021 Maxar (http://www.maxar.com/)
+ */
+
+#include "PolyToWayGeoModifierAction.h"
+
+// Hoot
+#include <hoot/core/util/Factory.h>
+
+using namespace std;
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(GeometryModifierAction, PolyToWayGeoModifierAction)
+
+const QString PolyToWayGeoModifierAction::NOOP_PARAM = "nothing";
+
+bool PolyToWayGeoModifierAction::processElement(const ElementPtr& pElement, OsmMap* pMap)
+{
+  // NOTE: The need for this code been met by a combination of AreaCriterion, TagCriterion and a SetTagVisitor
```

Comment:
If this code works best and its what we're going to use, I don't see the need for leaving this comment in.

Reply:
The need is to be able to export line and polygon features to a geodatabase as lines. Hoot is reading some features and deciding that they are polygons so they are unable to be saved as lines. This code is one of the options for converting from polygon to line.
Using the same 726Mb OSM file, the `GeometryModifierOp` was 25 seconds faster.

Option 1: Use the `GeometryModifierOp` to modify the geometry. 
```
conf file:
{
  "#" : "Export options",
  "convert.ops" : "$(convert.ops);AddUserIdVisitor;GeometryModifierOp",
  "geometry.modifier.rules.file" : "${HOOT_HOME}/translations/ModRules.json",
  "#" : "end"
}

ModRules.json file:
{
    "poly_to_way": {
        "filter": {
            "must": [{"tag": "natural=coastline"} ]
             },
         "arguments": {} }
}
```

Option 2:  Criterion & Visitors.
```
Conf file:
{
  "#" : "Export options",
  "convert.ops" : "$(convert.ops);AddUserIdVisitor;SetTagValueVisitor",
  "set.tag.value.visitor.chain.element.criteria" : "true",
  "set.tag.value.visitor.element.criteria":"AreaCriterion;TagCriterion",
  "tag.criterion.kvps" : "natural=coastline",
  "set.tag.value.visitor.keys" : "area",
  "set.tag.value.visitor.values" : "no",
  "#" : "end"
}
```